### PR TITLE
Warnings and errors for incompatible use of schema instance and many in fields.Nested

### PR DESCRIPTION
**Pull request by mmulqueen**
Addresses the issue I've described in #1982. Behaviour should be unchanged, but more obvious.

### CODE REVIEW

File: src/marshmallow/fields.py [hunk 1]
```diff
@@ -634,9 +640,14 @@ def _serialize(self, nested_obj, attr, obj, **kwargs):
         return schema.dump(nested_obj, many=many)
 
     def _test_collection(self, value):
+        if self.many and not self.schema.many:
+            raise FieldConfigurationError(
+                '"many" may not be used in conjunction with an instance in a nested field.'
+            )
```
Comment (lafrech) [inline src/marshmallow/fields.py:646]: I'd remove this test. This should be coverered by the init warning you added above, right?

File: src/marshmallow/fields.py [hunk 2]
```diff
@@ -634,9 +640,14 @@ def _serialize(self, nested_obj, attr, obj, **kwargs):
...
         many = self.schema.many or self.many
         if many and not utils.is_collection(value):
             raise self.make_error("type", input=value, type=value.__class__.__name__)
+        # Check whether many on the nested field and the schema are in conflict.
```
Comment (lafrech) [inline src/marshmallow/fields.py:650]: I guess this comment was meant for the change above?

File: src/marshmallow/fields.py [hunk 3]
```diff
@@ -551,6 +552,11 @@ def __init__(
                 "Use `Nested(lambda: MySchema(...))` instead.",
                 RemovedInMarshmallow4Warning,
             )
+        if isinstance(nested, SchemaABC) and many:
+            warnings.warn(
+                'Passing "many" is ignored if using an instance of a schema. Consider fields.List(fields.Nested(...)) instead.',
+                FieldConfigurationWarning,
+            )
```
Comment (lafrech) [inline src/marshmallow/fields.py:559]: I'd remove the suggestion.

**Review by lafrech — COMMENTED** [review]
Thanks for the contribution and sorry about the delay.

I understand the issue. Yet I'm reluctant to add code just to warn about this specific case.

I left a few comments anyway. I might change my mind and accept the test at init. No strong opinion against it.

**Comment by deckar01** [comment]
#779 lists other unexpected edge cases like this. I would rather bite the bullet and deprecate `many=True` than go down the path of maintaining edge case warnings.

**Comment by sloria** [comment]
closing as this has gone stale, and i agree with @deckar01 that we should just get rid of the `many=True` API altogether (eventually)
